core/src/main/scala-2/caliban/schema/ArgBuilderDerivation.scala

```diff
@@ -53,6 +56,27 @@ trait CommonArgBuilderDerivation {
         }
       case None        => Left(ExecutionError(s"Can't build a trait from input $input"))
     }
+
+  private def makeOneOfBuilder[A](ctx: SealedTrait[ArgBuilder, A]): ArgBuilder[A] =
+    new ArgBuilder[A] {
+
+      private def inputError(input: InputValue) =
+        ExecutionError(s"Invalid oneOf input $input for trait ${ctx.typeName.short}")
+
+      private val combined = ctx.subtypes.map(_.typeclass).toList.asInstanceOf[List[ArgBuilder[A]]] match {
+        case head :: tail =>
+          tail.foldLeft(head)(_ orElse _).orElse(input => Left(inputError(input)))
```

Comment:
This isn’t super nice, maybe we could have ArgBuilder have a partial function that can select if it can handle a particular input function. Then by default it would just be `def partial: PartialFunction[InputValue, Either[ExecutionError, A]] = PartialFunction(build)`. That way we could delegate to the ArgBuilder Wdyt?

Reply:
I gave it a go implementing your recommendation. Let me know what you think